### ISSUE TITLE
Respect `DOCKER_HOST` environment variable

### ISSUE THREAD

**Issue by gasinvein** [vv9k/dockeye#61]
#### Describe your feature request

Newer Podman versions support REST API over a unix socket, largely compatible with Docker. It would be nice if Dockeye respected `DOCKER_HOST` environment variable (instead of hardcoded `/run/docker.sock` path), so it could be at least tried to use with Podman (and who knows, maybe it will just work).

Podman sockets are at `/run/podman/podman.sock` (for system-wise rootful instance) and at `$XDG_RUNTIME_DIR/podman/podman.sock` (for per-user rootless instance). Dockeye could acknowledge and try these paths, but I believe respecting `DOCKER_HOST` would be still useful.

**Comment by vv9k** [comment]
Thanks for your interest in dockeye! The path is not hardcoded and can be changed in the setting. It was implemented in #23 (issue #22).

**Comment by gasinvein** [comment]
Oh, sorry for bothering, then. Still, maybe read `DOCKER_HOST` variable in a addition to the settings option?

**Comment by vv9k** [comment]
I'm not sure how that would work. If the environment variable is present which one should be preferred - the one in settings or the `DOCKER_HOST` variable?  I could perhaps see it implemented in such a way that first the configuration is read and then if the environment variable exists it is used instead.

Is the `DOCKER_HOST` variable a widely used thing? I don't see it exported on my machine.

**Comment by gasinvein** [comment]
> which one should be preferred - the one in settings or the `DOCKER_HOST` variable?

I think the `DOCKER_HOST` should be preferred if both are set, but I don't have a strong opinion on that.

> Is the `DOCKER_HOST` variable a widely used thing?

It's the conventional way to tell what Docker server the client should connect to (which includes changing the socket path for the local server). It's used by `docker` command and some other clients, e.g. the Docker UI extension for VSCode.

**Comment by vv9k** [comment]
I'll look into it then. Thanks for your suggestion!

**Comment by vv9k** [comment]
I added the necessary changes. Right now if the option in the settings to use `DOCKER_HOST` env var is checked and the variable exists, Dockeye will use it rather than the one defined in settings. I tested it out and it seems to detect it correctly. Next release should include this changes :)

I also tested out dockeye with podman and fixed some issues in the docker api layer so it should work more reliably. I plan to target Podman as the main platform after the basic docker api is finished.
